mlflow/keras.py

```diff
@@ -434,21 +440,31 @@ def fit(self, *args, **kwargs):
         unlogged_params = ['self', 'x', 'y', 'callbacks', 'validation_data', 'verbose']
 
         log_fn_args_as_params(original, args, kwargs, unlogged_params)
+        early_stop_callback = None
 
         # Checking if the 'callback' argument of fit() is set
         if len(args) >= 6:
             tmp_list = list(args)
+            early_stop_callback = early_stop_check(tmp_list[5])
             tmp_list[5] += [__MLflowKerasCallback()]
             args = tuple(tmp_list)
         elif 'callbacks' in kwargs:
+            early_stop_callback = early_stop_check(kwargs['callbacks'])
             kwargs['callbacks'] += [__MLflowKerasCallback()]
         else:
             kwargs['callbacks'] = [__MLflowKerasCallback()]
 
-        result = original(self, *args, **kwargs)
+        history = original(self, *args, **kwargs)
+
+        if early_stop_callback and early_stop_callback.restore_best_weights:
+            stopped_epoch_metrics = {key: history.history[key][early_stop_callback.stopped_epoch]
+                                     for key in history.history.keys()}
+            last_epoch = len(history.history['loss'])
```

Comment:
Let's be defensive here in case 'loss' is absent
```suggestion
            loss_history = history.history.get('loss')
            if loss_history is None:
                continue
            last_epoch = len(loss_history)
            ...
```

Or could we just use one of the metrics that we know for sure exists (e.g. pick a key from `history.history.keys()`)?

Reply:
Good catch, meant to change it forgot to go back. Will pick a key that exists.